csrank/callbacks.py

```diff
@@ -3,50 +3,95 @@
+
 
+class EarlyStoppingWithWeights(Callback, Tunable):
 
-class EarlyStoppingWithWeights(EarlyStopping, Tunable):
-    """Stop training when a monitored quantity has stopped improving.
-    # Arguments
-        monitor: quantity to be monitored.
-        min_delta: minimum change in the monitored quantity
-            to qualify as an improvement, i.e. an absolute
-            change of less than min_delta, will count as no
-            improvement.
-        patience: number of epochs with no improvement
-            after which training will be stopped.
-        verbose: verbosity mode.
-        mode: one of {auto, min, max}. In `min` mode,
-            training will stop when the quantity
-            monitored has stopped decreasing; in `max`
-            mode it will stop when the quantity
-            monitored has stopped increasing; in `auto`
-            mode, the direction is automatically inferred
-            from the name of the monitored quantity.
-    """
-
-    def __init__(self, **kwargs):
-        super(EarlyStoppingWithWeights, self).__init__(**kwargs)
+    def __init__(self, monitor='val_loss', min_delta=0, patience=0, verbose=0, mode='auto', baseline=None,
+                 restore_best_weights=False, **kwargs):
+        """
+            Stop training when a monitored quantity has stopped improving.
+
+            Parameters
+            ----------
+            monitor: string
+                Quantity to be monitored, could be the loss or an accuracy value, which is monitored by the model.
+                In case of accuracy the we check the change in increase, while in case of loss we check the change in
+                decrease of the loss
+            min_delta: float
+                Minimum change in the monitored quantity to qualify as an improvement, i.e. an absolute change of less
+                than min_delta, will count as no improvement
+            patience: unsigned int
+                number of epochs with no improvement after which training will be stopped
+            verbose : bool
+                verbosity mode, 1: print, 0 to not
+            mode: one of {auto, min, max}.
+                In `min` mode, training will stop when the quantity monitored has stopped decreasing; in `max` mode
+                it will stop when the quantity monitored has stopped increasing; in `auto` mode, the direction is
+                automatically inferred from the name of the monitored quantity
+            baseline: float
+                Baseline value for the monitored quantity to reach. Training will stop if the model doesn't show
+                improvement over the baseline
+            restore_best_weights: bool
+                whether to restore model weights from the epoch with the best value of the monitored quantity.
+                If False, the model weights obtained at the last step of training are used
+            **kwargs
+                Keyword arguments for the callback
```

Comment:
Remove one tabstop here to be consistent with the [numpy docstring style](https://numpydoc.readthedocs.io/en/latest/format.html).
Same thing applies to all other docstrings.